app/components/search_preview/search_preview.js

```diff
@@ -52,54 +58,89 @@ export default class SearchPreview extends PureComponent {
     };
 
     state = {
-        showPosts: false,
-        animationEnded: false
+        show: false,
+        error: false
     };
 
-    componentWillReceiveProps(nextProps) {
-        const {animationEnded, showPosts} = this.state;
-        if (animationEnded && !showPosts && nextProps.postIds.length) {
-            this.setState({showPosts: true});
+    constructor(props) {
+        super(props);
+
+        const {postIds} = props;
+        if (postIds && postIds.length >= 10) {
+            this.state.show = true;
         }
     }
 
+    componentDidMount() {
+        if (!this.state.show) {
+            this.loadPosts();
+        }
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return this.state.show !== nextState.show;
```

Comment:
Shouldn't it rerender if the posts/theme/channel changes at all? I don't think this is needed